### ISSUE TITLE
Strange DPY-5002 error with executemany batcherrors

### ISSUE THREAD

**Issue by anthony-tuininga** [oracle/python-oracledb#262]
### Discussed in https://github.com/oracle/python-oracledb/discussions/261

<div type='discussions-op-text'>

<sup>Originally posted by **vinidmpereira** December  4, 2023</sup>
Hello all,

I'm currently having a weird problem in Airflow using OracleHook, i know this forum pertains to oracledb but i think this is where i need help.

So i've been writing a large migrating script  to unify multiple similar databases into a singular one, and in one of the migrations i've been hit with:

 `oracledb.exceptions.InternalError: DPY-5002: internal error: read integer of length 3 when expecting integer of no more than length 2`

And this error only occurs in one of the 40+ databases i'm currently migrating, it's such a weird error and i can't find anything about this specific error, theres another topic where someone is currently having a similar mistake but with a DPY-5003 error, and i don't know if they are connected.
</div>

**Comment by anthony-tuininga** [comment]
I was table to replicate this. The protocol only allows for 65535 errors. If you exceed this value you get this error in thin mode. In thick mode you get invalid results (the number of batch errors is simply truncated by looking at the lowest 16 bits of the value) -- except when the number of batch errors is a multiple of 65535, in which case you get ``ORA-38902: errors in array DML exceed 65535``. As such, this is a bug in the database and will need to be fixed there. I will, however, correct the strange error in thin mode and ensure a more meaningful error is raised!

The issue can be avoided by always ensuring that you never send more than 65535 rows to the database at a time when using batch errors.

A simple test case that demonstrates the problem:

```sql
create table issue_262 (
    IntCol number(9) not null,
    StringCol varchar2(50) not null,
    constraint TestTempTable_pk primary key (IntCol)
);
```
with this Python script:
```python
import oracledb

# oracledb.init_oracle_client()

conn = oracledb.connect("user/password@host/service_name")
cursor = conn.cursor()

data = [(1, None)] * (65537)
cursor.executemany("insert into issue_262 values (:1, :2)", data,
                   batcherrors=True)
errors = cursor.getbatcherrors()
print("found", len(errors), "batch errors")
```

**Comment by anthony-tuininga** [comment]
I have pushed a patch that should correct this bug. If you are able to [build from source](https://python-oracledb.readthedocs.io/en/latest/user_guide/installation.html#installing-from-source-code) you can verify that it works for you.

**Comment by anthony-tuininga** [comment]
The patch has been included in version 2.0.0 which was just released.
